deployments/rollup-bsn-demo/demo.sh

```diff
@@ -1,387 +1,145 @@
-        VERIFY_SIG_CMD="/bin/babylond --home /babylondhome q wasm contract-state smart $finalityContractAddr '$verify_msg' --output json"
-        VERIFY_SIG_OUTPUT=$(docker exec babylondnode0 /bin/sh -c "$VERIFY_SIG_CMD")
-        
-        # Check if data is not null and contains our FP
-        if echo "$VERIFY_SIG_OUTPUT" | jq -e '.data != null' >/dev/null && echo "$VERIFY_SIG_OUTPUT" | jq -r '.data[]' | grep -q "$sig_fp_pubkey_hex"; then
-            verification_success=true
-            echo "    ✅ Verification succeeded on attempt $verification_attempt"
-            break
-        else
-            echo "    → Attempt $verification_attempt failed, data: $(echo "$VERIFY_SIG_OUTPUT" | jq -r '.data')"
-            if [ $verification_attempt -lt 5 ]; then
-                echo "    → Waiting 5 seconds before retry..."
-                sleep 5
-            fi
-        fi
-    done
-    
-    if [ "$verification_success" = false ]; then
-        echo "    ❌ Block $block_height: Finality signature verification failed"
-        echo "    → Verification output: $VERIFY_SIG_OUTPUT"
-        echo "  💥 Finality signature verification failed - stopping batch processing"
-        echo "  📊 Final status: $successful_sigs/$num_finality_sigs blocks processed successfully before failure"
-        exit 1
-    else
-        ((successful_sigs++))
-        echo "    ✅ Block $block_height: Finality signature submitted and verified successfully"
-    fi
-    
-    # Add small delay to avoid overwhelming the system
-    sleep 2  # Reduced since we have longer delays in verification
-done
-
-echo ""
-echo "🎉 All $num_finality_sigs finality signatures processed successfully!"
-echo "  📊 Successfully processed blocks $start_height to $((start_height + num_finality_sigs - 1))"
+# Wait a few seconds for the container and fpd daemon to fully start
+sleep 5
 
-###############################
-# Demo Summary                #
-###############################
+# Run create-finality-provider inside babylon-fp container
+docker exec babylon-fp fpd create-finality-provider \
+  --daemon-address 127.0.0.1:45661 \
+  --chain-id chain-test \
+  --eots-pk "$babylon_eotsd_pk" \
+  --commission-rate 0.05 \
+  --commission-max-change-rate 0.01 \
+  --commission-max-rate 0.20 \
+  --key-name babylon-key \
+  --moniker "Babylon FP" \
+  --website "https://myfinalityprovider.com" \
+  --security-contact "security@myfinalityprovider.com" \
+  --details "finality provider for the Babylon network" \
+  --home /home/babylonFpHome
+
+# Restart babylon-fp container so it picks up keys and config
+echo "Restarting babylon-fp container..."
+docker restart babylon-fp
+
+# Print Babylon FP info and save output
+output=$(bash ./scripts/print-babylon-fp.sh)
+btc_pk=$(echo "$output" | tail -n +2 | jq -r '.[0].btc_pk')
+echo "Babylon FP BTC Public Key: $btc_pk"
+
+# Delegate BTC using the extracted btc_pk
+bash ./scripts/delegate-btc-babylon-fp.sh "$btc_pk"
```

Comment:
Should we multi stake to both FPs in the later step?

Reply:
I tried that initially, but delegatation was not active after 30 attempts (quite a few minutes) so i reverted and wanted to report.

Let me test again and increae wait time